devicetypes/smartthings/zigbee-lock-without-codes/zigbee-lock-without-codes.groovy

```diff
@@ -0,0 +1,257 @@
+				zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY_PERCENTAGE_REMAINING)
+	return cmds
+}
+
+def parse(String description) {
+	def result = null
+	if (description) {
+		if (description.startsWith('read attr -')) {
+			result = parseAttributeResponse(description)
+		} else {
+			result = parseCommandResponse(description)
+		}
+	}
+	return result
+}
+
+private def parseAttributeResponse(String description) {
+	Map descMap = zigbee.parseDescriptionAsMap(description)
+	log.debug "Executing parseAttributeResponse() with description map:- $descMap"
+	def result = []
+	Map responseMap = [:]
+	def clusterInt = descMap.clusterInt
+	def attrInt = descMap.attrInt
+	def deviceName = device.displayName
+	responseMap.data = deviceName
+
+	if (clusterInt == CLUSTER_POWER && attrInt == POWER_ATTR_BATTERY_PERCENTAGE_REMAINING) {
+		responseMap.name = "battery"
+
+		if (Integer.parseInt(descMap.value, 16) != 255) {
+			responseMap.value = Math.round(Integer.parseInt(descMap.value, 16) / 2)
+			responseMap.descriptionText = "Battery is at ${responseMap.value}%"
+		}
+		
+	} else if (clusterInt == CLUSTER_DOORLOCK && attrInt == DOORLOCK_ATTR_LOCKSTATE) {
+		def value = Integer.parseInt(descMap.value, 16)
+		responseMap.name = "lock"
+		if (value == 0) {
+			responseMap.value = "unknown"
+			responseMap.descriptionText = "Unknown state"
+		} else if (value == 1) {
+			log.debug "locked"
+			responseMap.value = "locked"
+			responseMap.descriptionText = "Locked"
+		} else if (value == 2) {
+			log.debug "unlocked"
+			responseMap.value = "unlocked"
+			responseMap.descriptionText = "Unlocked"
+		} else {
+			responseMap.value = "unknown"
+			responseMap.descriptionText = "Unknown state"
+		}
+	} else {
+		return null
+	}
+	result << createEvent(responseMap)
+	return result
+}
+
+private def parseCommandResponse(String description) {
+	Map descMap = zigbee.parseDescriptionAsMap(description)
+	log.debug "Executing parseCommandResponse() with description map:- $descMap"
+
+	def deviceName = device.displayName
+	def result = []
+	Map responseMap = [:]
+	def data = descMap.data
+	def cmd = descMap.commandInt
+	def clusterInt = descMap.clusterInt
+	responseMap.data = deviceName
+
+	if (clusterInt == CLUSTER_DOORLOCK && (cmd == DOORLOCK_CMD_LOCK_DOOR || cmd == DOORLOCK_CMD_UNLOCK_DOOR)) {
+		def cmdList = []
+		cmdList << "delay 4200"
+		cmdList << zigbee.readAttribute(CLUSTER_DOORLOCK, DOORLOCK_ATTR_LOCKSTATE).first()
+		result << response(cmdList)
+	} else if (clusterInt == CLUSTER_DOORLOCK && cmd == DOORLOCK_RESPONSE_OPERATION_EVENT) {
+		def eventSource = Integer.parseInt(data[0], 16)
+		def eventCode = Integer.parseInt(data[1], 16)
+
+		responseMap.name = "lock"
+		responseMap.displayed = true
+		responseMap.isStateChange = true
+
+		if (eventSource == 1) {
+			def desc = "command"
+			responseMap.data = "command"
+		} else if (eventSource == 2) {
+			def desc = "manually"
+			responseMap.data = "manual"
+		}
```

Comment:
based on our current zigbee lock DTH, we want `desc` to be either "manually" or ""

and I think we want `responseMap.data = [method: "manual"]` (or "command")

Reply:
Done